pkg/binding/binding.go

```diff
@@ -0,0 +1,217 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/redhat-developer/odo/pkg/binding/asker"
+	"github.com/redhat-developer/odo/pkg/binding/backend"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	"github.com/redhat-developer/odo/pkg/libdevfile"
+)
+
+type BindingClient struct {
+	// Backends
+	flagsBackend       *backend.FlagsBackend
+	interactiveBackend *backend.InteractiveBackend
+
+	// Clients
+	kubernetesClient kclient.ClientInterface
+}
+
+func NewBindingClient(kubernetesClient kclient.ClientInterface) *BindingClient {
+	// We create the asker client and the backends here and not at the CLI level, as we want to hide these details to the CLI
+	askerClient := asker.NewSurveyAsker()
+	return &BindingClient{
+		flagsBackend:       backend.NewFlagsBackend(),
+		interactiveBackend: backend.NewInteractiveBackend(askerClient),
+		kubernetesClient:   kubernetesClient,
+	}
+}
+
+// GetFlags gets the flag specific to init operation so that it can correctly decide on the backend to be used
+// It ignores all the flags except the ones specific to init operation, for e.g. verbosity flag
+func (o *BindingClient) GetFlags(flags map[string]string) map[string]string {
+	initFlags := map[string]string{}
+	for flag, value := range flags {
+		if flag == backend.FLAG_NAME || flag == backend.FLAG_SERVICE {
+			initFlags[flag] = value
+		}
+	}
+	return initFlags
+}
+
+// Validate calls Validate method of the adequate backend
+func (o *BindingClient) Validate(flags map[string]string) error {
+	var backend backend.CreateBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+	return backend.Validate(flags)
+}
+
+func (o *BindingClient) SelectServiceInstance(flags map[string]string, options []string, serviceMap map[string]servicebinding.Ref) (string, error) {
+	var backend backend.CreateBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+	return backend.SelectServiceInstance(flags, options, serviceMap)
+}
+
+func (o *BindingClient) AskBindingName(serviceName, componentName string, flags map[string]string) (string, error) {
+	var backend backend.CreateBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+	defaultName := fmt.Sprintf("%v-%v", componentName, strings.Split(serviceName, " ")[0])
+	return backend.AskBindingName(defaultName, flags)
+}
+
+func (o *BindingClient) AskBindAsFiles(flags map[string]string) (bool, error) {
+	var backend backend.CreateBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+	return backend.AskBindAsFiles(flags)
+}
+
+func (o *BindingClient) CreateBinding(serviceName string, bindingName string, bindAsFiles bool, obj parser.DevfileObj, serviceMap map[string]servicebinding.Ref, componentContext string) error {
+	// serviceName format is <name> (<kind>.<apigroup>)
```

Comment:
Could we find a structure in https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/schema/group_version.go to pass this service name in a more structured way?